main/src/main/java/cgeo/geocaching/EditWaypointActivity.java

```diff
@@ -391,11 +397,29 @@ public static ImmutableTriple<FormulaEditText, FormulaEditText, TextSpinner<Dist
         }
     }
 
+    private Set<String> getNeededVariablesForProjection() {
+        final Set<String> projectionVars = new HashSet<>();
+        final ProjectionType pt = projectionType.get();
+        if (pt == ProjectionType.BEARING) {
+            FormulaUtils.addNeededVariables(projectionVars, binding.projectionBearingDistance.getFormulaText());
```

Comment:
`FormulaUtils.addNeededVariables` will unnecessarily recompile the formula while it is already available in `FormulaEditText`. Suggestion: add a method `getNeededVars` to class `FormulaEditText` and use this method instead.

Reply:
done